### PR TITLE
Prevent two-finger swipe forard/back when scrolling on viewport

### LINKED ISSUE #2548: Horizontal scroll with two-finger conflicts with swipe back/forward
**Describe the issue you encountered:**
Horizontal scroll with two-finger conflicts with swipe back/forward

**Expected behavior:**

- Platform (Win7, 8, 10? macOS? Linux distro?): OS X

- Brave Version: 0.11.1 Beta 1

- Steps to reproduce:
    1. Go to https://hootsuite.com
    2. Create timelines as much as you have to scroll horizontally

- Screenshot if needed:

- Any related issues: #2400

### LINKED ISSUE #2577: Horizontal scroll with two-finger conflicts with swipe back/forward on viewport
**Did you search for similar issues before submitting this one?**
Yes

**Expected behavior:** No conflict

- Platform (Win7, 8, 10? macOS? Linux distro?): OSX

- Brave Version: 0.11.1 Beta 1

- Steps to reproduce:
    1. go to google.com and random search some words
    2. press `option + command + j` to make window small 
    3. then horizontal scroll the page

- Screenshot if needed:

- Any related issues: #2548

### PR DESCRIPTION
**Pull request by darkdh**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

fix #2577

### CODE REVIEW

**Comment by darkdh** [comment]
This commit can only prevent conflicts on non-inner scrollable element like
screen shot 2016-07-18 at 18 23 31

**Comment by bbondy** [comment]
does Chrome work any better?

**Comment by darkdh** [comment]
Chrome perform swipe forward/back based on reaching the edge of viewport.
I think we can try it this way.

File: js/components/main.js [hunk 1]
```diff
@@ -125,12 +126,15 @@ class Main extends ImmutableComponent {
...
+      isScroll = true
+    })
     ipc.on('scroll-touch-begin', function () {
       trackingFingers = true
       startTime = (new Date()).getTime()
     })
     ipc.on('scroll-touch-end', function () {
-      if (trackingFingers && Math.abs(yVelocity) < 1) {
+      if (!isScroll && trackingFingers && Math.abs(yVelocity) < 1) {
         if (xVelocity > 1.5) {
```
Comment (diracdeltas) [inline js/components/main.js]: I think you can change the xVelocity threshold to 1 or 0.5 to make it easier to go back/forward now

**Comment by diracdeltas** [comment]
good idea

File: js/components/main.js (hunk 1 above)
Comment (darkdh) [inline js/components/main.js]: Thanks yan!

**Comment by darkdh** [comment]
about 656d56d:
1. We may need another way to pass the result back other than console.log
2. The reason not use velocity is some chrome users may swipe back/forward like drawing a bow with visual effect indicator which we can add in the future
screen shot 2016-07-19 at 15 02 38
3. This commit still can not prevent conflict with inner scrollable element

**Comment by bbondy** [comment]
Is IPC possible here? Just wondering why the console.log hack. Is it more efficient for some reason?

**Comment by darkdh** [comment]
@bbondy , your are right. I should use IPC. I just came up with console.log from nowhere.

**Comment by bbondy** [comment]
looks great, thanks!

**Pull request merged**